nmpolicy/internal/resolver/resolver.go

```diff
@@ -80,34 +94,37 @@ func (r *resolver) resolveCaptureEntryName(captureEntryName string) (types.NMSta
 	if !ok {
 		return nil, fmt.Errorf("capture entry '%s' not found", captureEntryName)
 	}
+	r.currentNode = &captureASTEntry
 	var err error
 	capturedStateEntry = types.CapturedState{}
-	capturedStateEntry.State, err = r.resolveCaptureASTEntry(captureASTEntry)
+	capturedStateEntry.State, err = r.resolveCaptureASTEntry()
 	if err != nil {
 		return nil, err
 	}
 	r.capturedStates[captureEntryName] = capturedStateEntry
 	return capturedStateEntry.State, nil
 }
 
-func (r resolver) resolveCaptureASTEntry(captureASTEntry ast.Node) (types.NMState, error) {
-	if captureASTEntry.EqFilter != nil {
-		return r.resolveEqFilter(captureASTEntry.EqFilter)
-	} else if captureASTEntry.Replace != nil {
-		return r.resolveReplace(captureASTEntry.Replace)
+func (r resolver) resolveCaptureASTEntry() (types.NMState, error) {
+	if r.currentNode.EqFilter != nil {
+		return r.resolveEqFilter()
+	} else if r.currentNode.Replace != nil {
+		return r.resolveReplace()
 	}
-	return nil, fmt.Errorf("root node has unsupported operation : %+v", captureASTEntry)
+	return nil, fmt.Errorf("root node has unsupported operation : %+v", *r.currentNode)
```

Comment:
Let's not print `%+v` it is not readable for users. Let's add `String()` method to `ast.Node`. We can address this in a follow-up.

Reply:
This should be addressed as part of fix on #44